tests/test_controllers.py

```diff
@@ -0,0 +1,174 @@
+        card_hist = Cardhist()
+        db_committing_function(round,round2,card_hist)
+        
+        process_card(card,player,False,False)
+        
+        self.assertTrue(process_card(card2,player,False,False))
+        
+    def test_process_card_with_resource_cost(self):
+        game = Game(age=1, round=1)
+        user = User(email='a@a.com', name='test', password='testcase')
+        
+        lumberYard = Card('Lumber Yard', 3, 1, 'brown')
+        lumberYard.set_benefit_wood(1)
+        baths = Card('Baths', 3, 1, 'blue')
+        baths.set_cost_stone(1)
+        baths.set_benefit_points(3)
+        stockade = Card('Stockade', 3, 1, 'red')
+        stockade.set_cost_wood(1)
+        stockade.set_benefit_military(1)
+        
+        db_committing_function(user,game,lumberYard,baths,stockade)
+        player = Player(userId=user.id, gameId=game.id, name=user.name)
+        db_committing_function(player)
+        round = Round(cardId=lumberYard.id, playerId=player.id)
+        round2a = Round(cardId=baths.id, playerId=player.id, round=2)
+        round2b = Round(cardId=stockade.id, playerId=player.id, round=2)
+        card_hist = Cardhist()
+        db_committing_function(round,round2a,round2b,card_hist)
+        
+        process_card(lumberYard,player,False,False)
+        
+        #can't play this card
+        self.assertFalse(process_card(baths,player,False,False))
+        
+        #can play this one
+        self.assertTrue(process_card(stockade,player,False,False))
+    
+    def test_process_card_discard(self):
+        game = Game(age=1, round=1)
+        user = User(email='a@a.com', name='test', password='testcase')
+        
+        lumberYard = Card('Lumber Yard', 3, 1, 'brown')
+        lumberYard.set_benefit_wood(1)
+        baths = Card('Baths', 3, 1, 'blue')
+        baths.set_cost_stone(1)
+        baths.set_benefit_points(3)
+        stockade = Card('Stockade', 3, 1, 'red')
+        stockade.set_cost_wood(1)
+        stockade.set_benefit_military(1)
+        
+        db_committing_function(user,game,lumberYard,baths,stockade)
+        
+        player = Player(userId=user.id, gameId=game.id, name=user.name)
+        db_committing_function(player)
+        round1 = Round(cardId=lumberYard.id, playerId=player.id)
+        round2 = Round(cardId=baths.id, playerId=player.id, round=2)
+        round3 = Round(cardId=stockade.id, playerId=player.id, round=3)
+        card_hist = Cardhist()
+        db_committing_function(round1,round2,round3,card_hist)
+        
+        process_card(lumberYard,player,False,False)
+        
+        #can play any cards regardless of whether they meet the resource cost
+        self.assertTrue(process_card(baths,player,True,False))
+        
+        round3 = Round(cardId=stockade.id, playerId=player.id, round=3)
+        db_committing_function(round3)
+        self.assertTrue(process_card(stockade,player,True,False))

```

Comment:
It would also be worth asserting that the player's money increases by 3 when they discard, and in the above test, that their other resources also change with the played cards.